app/Http/Controllers/Api/V1/TeamController.php

```diff
@@ -259,6 +261,75 @@ public function inviteMember(TeamInviteMemberRequest $inviteMemberRequest, Team
         ], 500);
     }
 
+    /**
+     * Invite Team Members
+     *
+     * Invite a bundle of users to the team.
+     *
+     * @bodyParam users array required The array of the users Example: [{id: 1, first_name: Jean, last_name: Erik, name: "Jean Erik"}, {id: "Kairo@Seed.com", email: "Kairo@Seed.com"}]
+     *
+     * @response {
+     *   "message": "Users have been invited to the team successfully."
+     * }
+     *
+     * @response 403 {
+     *   "errors": [
+     *     "You do not have permission to invite users to the team."
+     *   ]
+     * }
+     *
+     * @response 500 {
+     *   "errors": [
+     *     "Failed to invite users to the team."
+     *   ]
+     * }
+     *
+     * @param TeamInviteMembersRequest $inviteMembersRequest
+     * @param Team $team
+     * @return Response
+     */
+    public function inviteMembers(TeamInviteMembersRequest $inviteMembersRequest, Team $team)
+    {
+        $data = $inviteMembersRequest->validated();
+        $auth_user = auth()->user();
+        $owner = $team->getUserAttribute();
+
+        if ($owner->id === $auth_user->id) {
+            $invited = true;
+            foreach ($data['users'] as $user) {
+                $conUser = $this->userRepository->findByField('email', $user['email']);
+                $note = array_key_exists('note', $data) ? $data['note'] : '';
+
+                if ($conUser) {
+                    $token = Hash::make((string)Str::uuid() . date('D M d, Y G:i'));
+                    $team->users()->attach($conUser, ['role' => 'collaborator', 'token' => $token]);
+                    $conUser->notify(new InviteToTeamNotification($auth_user, $team, $token, $note));
+                } elseif($user['email']) {
+                    // TODO: need to send mail to unregistered person and do whatever for that person
+                    $token = Hash::make((string)Str::uuid() . date('D M d, Y G:i'));
+                    $tempUser = new User(['email' => $user['email']]);
+                    $tempUser->notify(new InviteToTeamNotification($auth_user, $team, $token, $note));
+                } else {
+                    $invited = false;
+                }
```

Comment:
There is a problem.
So, as I see it, even if a single 'user' record either doesn't have 'email' or is not an existing user, our API will return failed to invite users.
So, is it agreed to show error even for a partial success?

Reply:
Let's approve for now. I will discuss Leo about that later.

Reply:
Acknowledged!